src/tls/mod.rs

```diff
@@ -7,13 +7,13 @@ use std::{fmt::Debug, net::SocketAddr, path::PathBuf};
 use tokio::net::TcpStream;
 use tokio_openssl::{HandshakeError, SslStream};
 
-#[cfg(feature = "sources-tls")]
+#[cfg(feature = "tls")]
 mod incoming;
```

Comment:
If `incoming` is behind the `flag`, would it make sense for `outgoing` to also be behind the flag?

Reply:
Sorry, I missed this comment earlier. I believe this feature is really only intended to be used by sources, hence the `incoming` module. Per Bruce's comment below, I think it might make sense to extract some of the source specific bits from this module and into `src/sources/util/tls.rs`.